src/controllers/EventsController.js

```diff
@@ -1,176 +1,115 @@
-  _keepAlive() {
-    if (((new Date()) - this._lastMessage) >= 9000) {
-      this._lastMessage = new Date();
-      this.response.write('\n');
-      this._checkSocket();
-    }
+  _closeStream(subscriptionID: string): Promise<void> {
+    return new Promise((resolve: () => void) => {
+      // TODO i'm not sure if we need all 4 listens here
+      this.request.on('close', () => {
+        this._eventManager.unsubscribe(subscriptionID);
+        resolve();
+      });
+      this.request.on('end', () => {
+        this._eventManager.unsubscribe(subscriptionID);
+        resolve();
+      });
+      this.response.on('finish', () => {
+        this._eventManager.unsubscribe(subscriptionID);
+        resolve();
+      });
+      this.response.on('end', () => {
+        this._eventManager.unsubscribe(subscriptionID);
+        resolve();
+      });
+    });
   }
 
-  _checkSocket() {
+  _pipeEvent(event: Event) {
     try {
-      if (!this._socket) {
-        this._cleanup();
-        return false;
-      }
-
-      if (this.response.socket.destroyed) {
-        logger.log('Socket destroyed, cleaning up Event listener');
-        this._cleanup();
-        return false;
-      }
-
-      return true;
+      this.response.write(`event: ${event.name} \n\n`);
+      this.response.write(`data: ${JSON.stringify(eventToApi(event))}\n\n`);
     } catch (error) {
-      logger.error(`pipeEvents - error checking socket ${error}`);
-      return false;
+      logger.error(`pipeEvents - write error: ${error}`);
+      throw error;
     }
   }
 
-  _writeEventGen() {
-    return (
-      name: string,
-      data: ?Object,
-      ttl: ?number,
-      publishedAt: ?Date,
-      coreid: ? string,
-    ) => {
-      // if (filterCoreId && (filterCoreId !== coreid)) {
-      //   return;
-      // }
-      if (!this._checkSocket()) {
-        return;
-      }
-
-      try {
-        this._lastMessage = new Date();
-
-        const eventData = {
-          coreid: coreid || null,
-          data: data || null,
-          published_at: publishedAt || null,
-          ttl: ttl || null,
-        };
-
-        this.response.write(`event: ${name} \n`);
-        this.response.write(`data: ${JSON.stringify(eventData)} \n\n`);
-      } catch (error) {
-        logger.error(`pipeEvents - write error: ${error}`);
-      }
-    };
-  }
-
-  _pipeEvents() {
-    try {
-      this.request.socket.setNoDelay();
-      console.log('headers sent', this.response.headersSent);
-      this.response.writeHead(200, {
-        'Cache-Control': 'no-cache',
-        Connection: 'keep-alive',
-        'Content-Type': 'text/event-stream',
-      });
-
-      console.log('headers sent', this.response.headersSent);
-
-      this.response.write(':ok\n\n');
+  @httpVerb('get')
+  @route('/v1/events/:eventName?')
+  @serverSentEvents()
+  async getEvents(eventName: ?string): Promise<*> {
+    const subscriptionID = this._eventManager.subscribe(
+      eventName,
+      this._pipeEvent.bind(this),
```

Comment:
Oh.. I get it ..that is because in routeConfig.js we reassign context `only` for our routeHandler functions...so I have to use bind() or _pipeEvent (response) => (event) => {}.